vtex-sites/base.store
# chore: update login button

## What's the purpose of this pull request?
Update the login button to redirect to login if the user isn't authenticated or to redirect to my account if it is.

## How it works? 
It gets the user info from session and updates the button accordingly. 
The faststore update that enables this get from session is in [this PR](https://github.com/vtex/faststore/pull/1139). 

## How to test it?
The login flow consists in redirecting to `vtexcommercestable` first (for more information about those integrations, see our [docs](https://faststore.dev/how-to-guides/platform-integration/vtex/hosting-a-faststore-vtex-website)) in a subdomain, when the login is completed the auth cookie is set on the root domain also - in the case of `base.store` the root domain is `vtexfaststore.com`. It means that when you test the login flow in a preview you'll get redirected back to https://vtexfaststore.com/ after the login and if you access the preview again you won't be authenticated because the cookie wasn't set there.
As a workaround, you can get the `VtexIdclientAutCookie_ACCOUNT` (eg `VtexIdclientAutCookie_storeframework`) and add it on the preview, for that you can use the [EditThisCookie](https://www.editthiscookie.com/) extension.

https://user-images.githubusercontent.com/19983991/157094151-ac065096-f0c3-4961-9b32-5e420da6e54b.mov
